src/wallet/rpcwallet.cpp

```diff
@@ -3934,6 +3938,92 @@ UniValue autocombinerewards(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+UniValue setautocombinethreshold(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.empty() || request.params.size() > 2)
+        throw std::runtime_error(
+            "setautocombinethreshold enable ( value )\n"
+            "\nThis will set the auto-combine threshold value.\n"
+            "\nWallet will automatically monitor for any coins with value below the threshold amount, and combine them if they reside with the same PIVX address\n"
+            "When auto-combine runs it will create a transaction, and therefore will be subject to transaction fees.\n"
+
+            "\nArguments:\n"
+            "1. enable          (boolean, required) Enable auto combine (true) or disable (false).\n"
+            "2. threshold       (numeric, optional. required if enable is true) Threshold amount. Must be greater than 1.\n"
+
+            "\nResult:\n"
+            "{\n"
+            "  \"enabled\": true|false,      (boolean) true if auto-combine is enabled, otherwise false\n"
+            "  \"threshold\": n.nnn,         (numeric) auto-combine threshold in PIV\n"
+            "  \"saved\": true|false         (boolean) true if setting was saved to the database, otherwise false\n"
+            "}\n"
+
+            "\nExamples:\n" +
+            HelpExampleCli("setautocombinethreshold", "500.12") + HelpExampleRpc("setautocombinethreshold", "500.12"));
+
+    EnsureWallet();
+
+    RPCTypeCheck(request.params, {UniValue::VBOOL, UniValue::VNUM});
+
+    bool fEnable = request.params[0].get_bool();
+
+    if (fEnable && request.params.size() < 2)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Missing threshold value");
+
+    CWalletDB walletdb(pwalletMain->GetDBHandle());
+    CAmount nThreshold = 0;
+
+    if (fEnable && request.params.size() > 1) {
```

Comment:
nit: no need to check the params size again, if we didn't throw at line 3971 we are good. 
We could refactor this a bit, and do both checks here, e.g.
```cpp
if (fEnable) {
    if (request.params.size() < 2) {
        throw JSONRPCError(RPC_INVALID_PARAMETER, "Missing threshold value");
    }
    nThreshold = AmountFromValue(request.params[1]);
    ...
}
```

Reply:
needed to rebase this anyways, so took care of this while i was at it